core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -28,18 +28,35 @@ oppia.directive('collectionNodeCreator', [
         'validatorsService', 'CollectionEditorStateService',
         'CollectionLinearizerService', 'CollectionUpdateService',
         'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-        'siteAnalyticsService',
+        'SearchExplorationsBackendApiService', 'siteAnalyticsService',
         function(
             $scope, $http, $window, $filter, alertsService,
             validatorsService, CollectionEditorStateService,
             CollectionLinearizerService, CollectionUpdateService,
             CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-            siteAnalyticsService) {
+            SearchExplorationsBackendApiService, siteAnalyticsService) {
           $scope.collection = CollectionEditorStateService.getCollection();
           $scope.newExplorationId = '';
           $scope.newExplorationTitle = '';
           var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
 
+          $scope.fetchExplorationsMetadata = function(searchQuery) {
+            if (/^[a-zA-Z0-9- ]*$/.test(searchQuery)) {
```

Comment:
Extract into isValidExplorationTitle() method; maybe use the validator service in app.js. Make sure that the validation here is the same as any existing validation that's being used to validate the charsets for exploration titles, so that if the allowed charset constant changes then this automatically changes too.